core/src/main/java/com/linecorp/armeria/common/util/AbstractOptions.java

```diff
@@ -188,37 +201,14 @@ protected AbstractOptions(AbstractOptions baseOptions, AbstractOptions additiona
         return optionValue != null ? optionValue.value() : null;
     }
 
-    /**
-     * Returns the value of the specified {@code option}.
-     *
-     * @param <O> the type of the option
-     * @param <V> the type of the value
-     * @return the value of the specified {@code option}. {@code defaultValue} if there's no such option.
-     */
-    protected final <O extends AbstractOption<V>, V> V getOrElse0(O option, V defaultValue) {
-        requireNonNull(defaultValue, "defaultValue");
-        final V value = getOrNull0(option);
-        if (value != null) {
-            return value;
-        } else {
-            return defaultValue;
-        }
-    }
-
-    /**
-     * Returns the {@link Map} whose key is {@link AbstractOption} and value is {@link AbstractOptionValue}.
-     *
-     * @param <K> the type of the options
-     * @param <V> the type of the option values
-     */
-    @SuppressWarnings("unchecked")
-    protected final <K extends AbstractOption<?>, V extends AbstractOptionValue<K, ?>> Map<K, V> asMap0() {
-        return Collections.unmodifiableMap((Map<? extends K, ? extends V>) valueMap);
+    @Override
+    public Iterator<T> iterator() {
+        return Iterators.unmodifiableIterator(valueMap.values().iterator());
     }
 
     @Override
     public String toString() {
-        return toString(asMap0().values());
+        return toString(valueMap.values());
     }
 
     static String toString(Collection<?> values) {
```

Comment:
Not necessarily part of this PR, but how about rewriting this method with `MoreObjects.toStringHelper`?